react/src/lib/components/SubsurfaceViewer/layers/wells/wellsLayer.ts

```diff
@@ -298,10 +302,14 @@ export default class WellsLayer extends CompositeLayer<
         }
 
         const refine = this.props.refine;
-        const data = refine
+        let data = refine
             ? splineRefine(this.props.data as unknown as FeatureCollection) // smooth well paths.
             : (this.props.data as unknown as FeatureCollection);
 
+        if (!this.props.isZDepth) {
```

Comment:
Not sure if it makes any different, but maybe it's more efficient to invert before refining (less samples to invert)?